File: Week_2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int get_grade(int letters, int words, int sentences);

int main(void)
{
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    int grade = get_grade(letters, words, sentences);
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", grade);
    }
    return (0);
}

int count_letters(string text)
{
    int i = 0;
    int letters = 0;
    while (text[i])
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }
        i++;
    }
    return (letters);
}

int count_words(string text)
{
    int i = 0;
    int words = 1;
    while (text[i])
    {
        if (text[i + 1] == ' ')
        {
            words++;
        }
        i++;
    }
    return (words);
}

int count_sentences(string text)
{
    int i = 0;
    int sentences = 0;
    while (text[i])
    {
        if (text[i] == '!' || text[i] == '.' || text[i] == '?')
        {
            sentences++;
        }
        i++;
    }
    return (sentences);
}


int get_grade(int letters, int words, int sentences)
{
    float grade = 0;
    float l = 0;
    l = (float)letters / (float)words * 100;
    float s = (float)sentences / (float)words * 100;
    grade = round(0.0588 * l - 0.296 * s - 15.8);
    return (grade);
}

File: Week_6/sentimental-readability/readability.py
# TODO
from cs50 import get_string


def count_letters(text):
    letters = 0
    for element in text:
        if ((element >= 'a' and element <= 'z') or (element >= 'A' and element <= 'Z')):
            letters += 1
    return (letters)


def count_words(text):
    words = 1
    for word in text:
        if (word == ' '):
            words += 1
    return (words)


def count_sentences(text):
    sentences = 0
    for el in text:
        if (el == '!' or el == '.' or el == '?'):
            sentences += 1
    return (sentences)


def get_grade(letters, words, sentences):
    grade = 0
    l = 0
    l = letters / words * 100
    s = sentences / words * 100
    grade = round(0.0588 * l - 0.296 * s - 15.8)
    return int(grade)


text = get_string("Text: ")
letters = count_letters(text)
words = count_words(text)
sentences = count_sentences(text)
grade = get_grade(letters, words, sentences)
if (grade < 1):
    print("Before Grade 1")
elif (grade >= 16):
    print("Grade 16+")
else:
    print(f"Grade {grade}")